File: lib/src/main/java/lib/kalu/adapter/animation/SlideInLeftAnimation.java
package lib.kalu.adapter.animation;

import android.animation.Animator;
import android.animation.ObjectAnimator;
import android.view.View;

public final class SlideInLeftAnimation implements BaseAnimation {

    @Override
    public Animator[] getAnimators(View view) {

        final ObjectAnimator animator = ObjectAnimator.ofFloat(view, "translationX", -view.getRootView().getWidth(), 0);
        return new Animator[]{animator};
    }
}

File: lib_kotlin/src/main/java/lib/kalu/adapter/animation/SlideInLeftAnimation.kt
package lib.kalu.adapter.animation

import android.animation.Animator
import android.animation.ObjectAnimator
import android.view.View

class SlideInLeftAnimation : BaseAnimation {

    override fun getAnimators(view: View): Array<Animator> {
        return arrayOf(ObjectAnimator.ofFloat(view, "translationX", -view.rootView.width.toFloat(), 0f))
    }
}
